dbt-tests-adapter/dbt/tests/adapter/incremental/test_incremental_on_schema_change.py

```diff
@@ -84,3 +84,27 @@ def test_run_incremental_fail_on_schema_change(self, project):
 
 class TestIncrementalOnSchemaChange(BaseIncrementalOnSchemaChange):
     pass
+
+
+class BaseIncrementalCaseSenstivityOnSchemaChange:
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "src_artists.sql": fixtures._MODELS__SRC_ARTISTS,
+            "dim_artists.sql": fixtures._MODELS__DIM_ARTISTS,
+        }
+
+    def test_run_incremental_check_quoting_on_new_columns(self, project):
+        select = "src_artists dim_artists"
+        run_dbt(["run", "--models", select, "--full-refresh"])
+        run_dbt(["show", "--inline", "select * from {{ ref('dim_artists') }}"])
+        res, logs = run_dbt_and_capture(
+            ["show", "--inline", "select * from {{ ref('dim_artists') }}"]
+        )
+        assert "Job" not in logs
+        run_dbt(["run", "--vars", "{'version': 1}"])
+        run_dbt(["show", "--inline", "select * from {{ ref('dim_artists') }}"])
+        res, logs = run_dbt_and_capture(
+            ["show", "--inline", "select * from {{ ref('dim_artists') }}"]
+        )
+        assert "Job" in logs
```

Comment:
The `run_dbt(["show" ...` portions probably aren't necessary. They were just so people could see the state of the data within this reprex: https://github.com/dbt-labs/dbt-adapters/issues/250#issuecomment-2192228487

Instead of the assertions here, you might be able to just rely upon `expect_pass=True` as the default for [`run_dbt`](https://github.com/dbt-labs/dbt-core/blob/014444dc18dd18ab75b81ad62530256ec09df22a/core/dbt/tests/util.py#L73-L75) and [`run_dbt_and_capture`](https://github.com/dbt-labs/dbt-core/blob/014444dc18dd18ab75b81ad62530256ec09df22a/core/dbt/tests/util.py#L111-L113).

From the original bug report in https://github.com/dbt-labs/dbt-adapters/issues/250, my understanding was that it actually produced an error.